### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by phatnhse** [TienVNguyen/ToDoAppProject#1]
### Thank you for your submission TienVNguyen.
- You really pushed effort on doing the prework, I really appreciate that.
- We will send you an email soon to accept your submission.
### Good stuffs
- Your code is beautiful. It organized accordingly with best practices into sub-packages. The way you named your variables, properties, methods make me feel like you have played around with this language for a long time.
- Using `CustomAdapter` and `ViewHolder` pattern is great.It makes your app run with better performance.
- Good use `SQlite` with `DatabaseConstant`.
- Nice to see you use many `View` types likes `DatePickerDialog`, `TimePickerDialog`, `ProgressBar`, `SeekBar`... It makes your application really Materialize
- `ButterKnife` is absolutely a useful library.Good to see you use it.
- Nice use `PlashScreen`.
- Using `EmptyView` pattern and `ProgressBar` before loading data is cool. Better UX!
### Suggestions
- Seem like `AddActivity` is similar to `EditOrRemoveActivity`. It's hard to maintain later. How about creating a BaseActivity and extend from it? Or 1 activity with FLAG_CREATE and FLAG_EDIT, you can handle it easily. (Recommended the first one)
- I suggest using `RecyclerView` instead of `ListView`. 
- Using `DialogFragment` instead of creating a new `Activity` is a great way to edit item. 
- I'm very happy to see you played with multiple `View` types. But in some cases, about the UX, it has a little bit hard to follow. Ex: Status/Percent/Completed is multiple ways to show the progress of item ?
  
  In general, your code is awesome. 
### Questions
- How long have you developed Android? 
- What are you expecting for in this course?

**Comment by TienVNguyen** [comment]
Hi @lazybee27102,

It's great to receive your feedback, especially the suggestions. So, I will update mine as much as I can.

For your question,
- How long have you developed Android?
- I have learned Android through Online resources (Youtube, Udacity, Codepath) for about 5 months (around 3 hours per day).
- What are you expecting for in this course?
- I love to learn lots of things about Android. There still be many hard things to understand to implement like RecyclerView, Fragment, etc. And I want to improve my experience through suggestions.
- I like to meet many developers who care about Android.

Thanks & Best Regards,
Tien Nguyen

**Comment by phatnhse** [comment]
Thanks for your response @TienVNguyen

Spending 3 hours for a stuff everyday isn't easy. I am appreciated when hearing that from you. Keep moving up. Hope to see your update soon. 

I'm here to be helpful. If you have any question, please tell me know.

**Comment by TienVNguyen** [comment]
Hi @lazybee27102,

I just applied the basic of both RecyclerView (to List) and DialogFragment (to Filter option). It's a great to adapt new things.

Thanks & Best Regards,
Tien Nguyen

**Comment by TienVNguyen** [comment]
This issue is fixed. And it is ready to close.

Thanks & Best Regards,
Tien Nguyen
